### ISSUE TITLE
Link to the same existing attachment instead of storing a new file

### ISSUE THREAD

**Issue by tkdchen** [Nitrate/Nitrate#570]
When a user is uploading a file that has same content as an existing one in the server side, Nitrate should link the uploaded file to the existing one rather than creating and storing a new file under the upload directory in the server.

AC:
* A new attachment object is created
* The new attachment object should have a field pointing to another attachment object that has same file content.

Start after #569

**Comment by tkdchen** [comment]
Deferred. Reopen if this is necessary in the future.
